Tests/NetworkServiceTests/NetworkServiceTests.swift

```diff
@@ -1,8 +1,114 @@
-import NetworkService
+@testable import NetworkService
 import XCTest
 
 final class NetworkServiceTests: XCTestCase {
-    func testUseMethodExistsInAPI() {
-        NetworkService().testUse()
+    struct MockEndpoint: HTTPAPIEndpoint {
+        static var networkEnvironment: String = "test"
+        var baseURL: URL = .init(string: "https://api.example.com")!
+        var path: String = "/endpoint"
+        var task: HTTPTask = .request
+        var method: HTTPMethod = .get
+        var headers: HTTPHeaders = [:]
+    }
```

Comment:
could we move this to a separate file, because it looks messy to me?

Reply:
Separated to another file with 203e327